src/main/java/duke/command/ShowAllTasksOnSameDateCommand.java

```diff
@@ -0,0 +1,118 @@
+package duke.command;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.utils.CortanaException;
+import duke.utils.Storage;
+import duke.utils.TaskList;
+import duke.utils.Ui;
+
+/**
+ * The type Show all tasks on same date command.
+ */
+public class ShowAllTasksOnSameDateCommand extends Command {
+    private final LocalDateTime dateTime;
+    private final String dateTimeString;
+
+    /**
+     * Instantiates a new Show all tasks on same date command.
+     *
+     * @param dateTime       the specified date time
+     * @param dateTimeString the specified date time in string
+     */
+    public ShowAllTasksOnSameDateCommand(LocalDateTime dateTime, String dateTimeString) {
+        this.dateTime = dateTime;
+        this.dateTimeString = dateTimeString;
+    }
+
+    /**
+     * Execute show all tasks on the same date/time operation.
+     *
+     * @param taskList the task list to operate on
+     * @param ui the ui to operate on
+     * @param storage the storage to operate on
+     */
+    public String execute(TaskList taskList, Ui ui, Storage storage) throws CortanaException {
```

Comment:
Would this be a '[deep nesting](https://nus-cs2103-ay2122s2.github.io/website/se-book-adapted/chapters/codeQuality.html)' ?

Reply:
I agree with @tzhan98. Perhaps you could consider the suggestions given [here](https://blog.codinghorror.com/flattening-arrow-code/)?